### PR TITLE
Update of the total balance in bg thread.

### PR DESCRIPTION
**Pull request by OlegGordiichuk**
Calculation of the total balance will be done in bg thread.

### CODE REVIEW

File: Trust/Tokens/ViewModels/TokensViewModel.swift [hunk 1]
```diff
@@ -85,20 +81,25 @@ final class TokensViewModel: NSObject {
...
-        guard let coinTicker = store.coinTicker(by: token.address) else {
-            return 0
+    func amount(completion: @escaping (String?) -> Void) {
+        DispatchQueue.global(qos: .background).async { [weak self] in
+          guard let strongSelf = self else {
+                completion(CurrencyFormatter.formatter.string(from: NSNumber(value: TokenObject.DEFAULT_BALANCE)))
+                return
+          }
+
+          let realm = try! Realm(configuration: strongSelf.store.realm.configuration)
```
Comment (vikmeup) [inline Trust/Tokens/ViewModels/TokensViewModel.swift]: what's the reason for initializing realm? Can you just `use store.realm`?

Comment (vikmeup) [reply Trust/Tokens/ViewModels/TokensViewModel.swift]: If it's shared, then can you pass separate instance?

File: Trust/Tokens/ViewControllers/TokensViewController.swift [hunk 2]
```diff
@@ -116,7 +115,9 @@ final class TokensViewController: UIViewController {
     }
 
     func refreshHeaderView() {
-        header.amountLabel.text = viewModel.headerBalance
+        viewModel.amount(completion: { [weak self] value in
```
Comment (vikmeup) [inline Trust/Tokens/ViewControllers/TokensViewController.swift]: would this update total amount on Token view screen?

File: Trust/Tokens/Views/TokensHeaderView.swift [hunk 3]
```diff
@@ -9,6 +9,7 @@ final class TokensHeaderView: UIView {
         let label = UILabel(frame: .zero)
         label.textColor = Colors.black
         label.textAlignment = .center
+        label.text = CurrencyFormatter.formatter.string(from: NSNumber(value: TokenObject.DEFAULT_BALANCE))
```
Comment (vikmeup) [inline Trust/Tokens/Views/TokensHeaderView.swift]: Can this be set from the `viewModel`?

File: Trust/Tokens/ViewModels/TokensViewModel.swift (hunk 1 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewModels/TokensViewModel.swift]: We should create separate Realm instance in specific bg thread.

File: Trust/Tokens/ViewControllers/TokensViewController.swift (hunk 2 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/ViewControllers/TokensViewController.swift]: Yeah it will

File: Trust/Tokens/Views/TokensHeaderView.swift (hunk 3 above)
Comment (OlegGordiichuk) [reply Trust/Tokens/Views/TokensHeaderView.swift]: done

File: Trust/Tokens/ViewModels/TokensViewModel.swift (hunk 1 above)
Comment (vikmeup) [reply Trust/Tokens/ViewModels/TokensViewModel.swift]: Can you remove `!` - I hope this won't happen, but ideally, we need to reduce the number of `!` in the code

**Pull request closed without merging**